### ISSUE TITLE
YOLOv10 training at ~10s/iteration using a custom dataset

### ISSUE THREAD

**Issue by CodeNiJD** [ultralytics/ultralytics#17414]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

Hi! I am very new at learning about object detection. I am trying to create a project that detects car license plates in videos. I came across YOLOv10 and followed some tutorials on youtube on how to use it. I am using a dataset I got from Roboflow which contains Philippine License Plates. However, I have noticed that GPU mem keeps showing 0G throughout all the epochs and the speed is around 10secs/iteration. I've been resuming training for a couple of days now but it's really taking a lot of my time.
![image](https://github.com/user-attachments/assets/4b18425c-3189-4f7e-aad4-c94bfa1bac7d)

I have also investigated about possible causes and I have found out that PyTorch does not have CUDA enabled and it seems like my GPU is not being used (I have a GTX1070). I tried to install torch with CUDA by following the selector in the PyTorch website but there were a lot of errors when I started training the model. 

Currently, I have reverted torch to the previous setting.

I still have very little idea on what is happening and anything would really help. Thanks in advance!

### Environment

Ultralytics YOLOv8.1.34 🚀 Python-3.10.15 torch-2.0.1+cpu CPU (Intel Core(TM) i7-6700 3.40GHz)
Setup complete ✅ (8 CPUs, 32.0 GB RAM, 867.4/931.4 GB disk)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.10.15
Install             pip
RAM                 31.95 GB
CPU                 Intel Core(TM) i7-6700 3.40GHz
CUDA                None

matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.0>=1.4.1
torch               ✅ 2.0.1>=1.8.0
torchvision         ✅ 0.15.2>=0.9.0
tqdm                ✅ 4.67.0>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0

### Minimal Reproducible Example

from ultralytics import YOLOv10

config_path = './config.yaml'

model = YOLOv10.from_pretrained("jameslahm/yolov10n")

model.train(data=config_path, epochs=50, batch=16)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @CodeNiJD, thank you for your interest in Ultralytics 🚀! We recommend visiting our [Docs](https://docs.ultralytics.com) to find extensive examples and tips, which might help address common questions and issues you may encounter as a new user.

It seems like you're experiencing a challenge related to GPU usage during training. If this is indeed a 🐛 Bug Report, please ensure you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us understand and debug the issue better.

Since it appears that CUDA might not be enabled on your PyTorch installation, double-check if you have a compatible CUDA version installed and try reinstalling PyTorch with CUDA support using the [PyTorch installation guide](https://pytorch.org/get-started/locally/). This will likely resolve the issue with your GPU not being utilized.

If your situation involves custom training ❓ Question, please provide more specifics, such as dataset samples and training logs. Additionally, reviewing our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) might offer useful insights.

For community support, you’re more than welcome to join the Ultralytics community on [Discord](https://ultralytics.com/discord) 🎧 for real-time assistance, explore discussions on [Discourse](https://community.ultralytics.com), or engage with us on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you're using the latest `ultralytics` package and all necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) by upgrading your installation:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be executed in any of these verified environments with pre-configured dependencies, including [CUDA](https://developer.nvidia.com/cuda), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU access: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Implement with Deep Learning VM as per [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: Utilize AWS Deep Learning AMI as directed in [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Refer to [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge indicates all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are successful. Our CI validates YOLO's correct operation across all [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and upon every commit.

Please note that this is an automated response, and an Ultralytics engineer will review and assist you with your issue shortly. If you have further questions or need additional help, our community and engineering teams are here to support you.

**Comment by Y-T-G** [comment]
You shouldn't simply say "there were errors" when you get errors if you're reporting errors without posting the error. It's slow because like you said it's not using CUDA. You should find your CUDA driver version by running `nvidia-smi` in command prompt and then find the PyTorch version with CUDA version that's lower to equal to the one that you see after running that command. Your GPU is too old to support the latest PyTorch version.

https://pytorch.org/get-started/previous-versions/
